clients/vault/src/metrics.rs

```diff
@@ -1,27 +1,1341 @@
+	let surplus = *vault.metrics.fee_budget_surplus.data.read().await;
+	vault
+		.metrics
+		.fee_budget_surplus
+		.gauge
+		.set(surplus as f64 / vault.vault_id.wrapped_currency().inner()?.one() as f64);
+	Ok(())
+}
+
+async fn publish_average_bitcoin_fee(vault: &VaultData) {
+	let average = match vault.metrics.average_btc_fee.data.read().await {
+		x if x.count > 0 => x.total as f64 / x.count as f64,
+		_ => Default::default(),
+	};
+	vault.metrics.average_btc_fee.gauge.set(average);
+}
+
+// fn publish_bitcoin_balance(vault: &VaultData) {
+//     match vault.btc_rpc.get_balance(None) {
+//         Ok(bitcoin_balance) => vault.metrics.btc_balance.actual.set(bitcoin_balance.to_btc() as
+// f64),         Err(e) => {
+//             // unexpected error, but not critical so just continue
+//             tracing::warn!("Failed to get Bitcoin balance: {}", e);
+//         }
+//     }
+// }
+
+async fn publish_native_currency_balance<P: CollateralBalancesPallet + UtilFuncs>(
+	parachain_rpc: &P,
+) -> Result<(), ServiceError<Error>> {
+	let native_currency = parachain_rpc.get_native_currency_id();
+	if let Ok(balance) = parachain_rpc.get_free_balance(native_currency).await {
+		let balance = raw_value_as_currency(balance, native_currency)?;
+		NATIVE_CURRENCY_BALANCE.set(balance);
+	}
+	Ok(())
+}
+
+// fn publish_utxo_count(vault: &VaultData) {
+//     if let Ok(count) = vault.btc_rpc.get_utxo_count() {
+//         if let Ok(count_i64) = count.try_into() {
+//             vault.metrics.utxo_count.set(count_i64);
+//         }
+//     }
+// }
+
 pub fn increment_restart_counter() {
 	RESTART_COUNT.inc();
 }
 
-pub async fn publish_tokio_metrics(
-	_metrics_iterators: HashMap<String, impl Iterator<Item = TaskMetrics>>,
+async fn publish_issue_count<V: VaultDataReader, P: IssuePallet + UtilFuncs>(
+	parachain_rpc: &P,
+	vault_id_manager: &V,
+) {
+	if let Ok(issues) = parachain_rpc
+		.get_vault_issue_requests(parachain_rpc.get_account_id().clone())
+		.await
+	{
+		for vault in vault_id_manager.get_entries().await {
+			let relevant_issues: Vec<_> = issues
+				.iter()
+				.filter(|(_, issue)| issue.vault == vault.vault_id)
+				.map(|(_, issue)| issue.status.clone())
+				.collect();
+
+			vault.metrics.issues.open_count.set(
+				relevant_issues
+					.iter()
+					.filter(|status| matches!(status, IssueRequestStatus::Pending))
+					.count() as f64,
+			);
+			vault.metrics.issues.completed_count.set(
+				relevant_issues
+					.iter()
+					.filter(|status| matches!(status, IssueRequestStatus::Completed))
+					.count() as f64,
+			);
+			vault.metrics.issues.expired_count.set(
+				relevant_issues
+					.iter()
+					.filter(|status| matches!(status, IssueRequestStatus::Cancelled))
+					.count() as f64,
+			);
+		}
+	}
+}
+
+async fn publish_time_to_first_deadline<V: VaultDataReader, P: RedeemPallet + SecurityPallet>(
+	parachain_rpc: &P,
+	vault_id_manager: &V,
+	redeems: &[(H256, SpacewalkReplaceRequest)],
+) {
+	todo!()
```

Comment:
Is this intended? So this part will be completed in another PR? or is this still a draft PR?

Reply:
so this function related to xlm network. so will add/modify logic in next pr.